tabular/src/autogluon/tabular/models/tabpfnv2/tabpfnv2_model.py

```diff
@@ -0,0 +1,356 @@
+            )
+        else:
+            if n_ensemble_repeats is not None:
+                hps["n_estimators"] = n_ensemble_repeats
+            self.model = model_base(**hps)
+
+        self.model = self.model.fit(
+            X=X,
+            y=y,
+        )
+
+    def _get_default_resources(self) -> tuple[int, int]:
+        num_cpus = ResourceManager.get_cpu_count_psutil()
+        num_gpus = min(ResourceManager.get_gpu_count_torch(), 1)
+        return num_cpus, num_gpus
+
+    def _set_default_params(self):
+        default_params = {
+            "random_state": 42,
+            "ignore_pretraining_limits": True,  # to ignore warnings and size limits
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    @classmethod
+    def supported_problem_types(cls) -> list[str] | None:
+        return ["binary", "multiclass", "regression"]
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        default_auxiliary_params.update(
+            {
+                "max_rows": 10000,
+                "max_features": 500,
+                "max_classes": 10,
+            }
+        )
+        return default_auxiliary_params
+
+    @classmethod
+    def _get_default_ag_args_ensemble(cls, **kwargs) -> dict:
+        """
+        Set fold_fitting_strategy to sequential_local,
+        as parallel folding crashes if model weights aren't pre-downloaded.
+        """
+        default_ag_args_ensemble = super()._get_default_ag_args_ensemble(**kwargs)
+        extra_ag_args_ensemble = {
+            # FIXME: Find a work-around to avoid crash if parallel and weights are not downloaded
+            "fold_fitting_strategy": "sequential_local",
+        }
+        default_ag_args_ensemble.update(extra_ag_args_ensemble)
+        return default_ag_args_ensemble
+
+    def _estimate_memory_usage(self, X: pd.DataFrame, **kwargs) -> int:
+        hyperparameters = self._get_model_params()
+        return self.estimate_memory_usage_static(
+            X=X,
+            problem_type=self.problem_type,
+            num_classes=self.num_classes,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+
+    @classmethod
+    def _estimate_memory_usage_static(
+        cls,
+        *,
+        X: pd.DataFrame,
+        hyperparameters: dict = None,
+        **kwargs,
+    ) -> int:
+        """
+        Heuristic memory estimate that is very primitive.
+        Can be vastly improved.
+        """
+        num_features = len(X.columns)
+        columns_mem_est = num_features * 1e6
+
+        dataset_size_mem_est = 10 * get_approximate_df_mem_usage(X).sum()  # roughly 10x DataFrame memory size
+        baseline_overhead_mem_est = 2e9  # 2 GB generic overhead
+
+        mem_estimate = dataset_size_mem_est + columns_mem_est + baseline_overhead_mem_est
+
+        return mem_estimate
```

Comment:
This memory estimate is likely very inaccurate. Better estimates are welcome!

Reply:
If I'm reading this right, that means TabPFNv2 memory usage scales linearly with the number of cells in X?

Reply:
I've updated the memory estimate to align with TabPFN's GPU memory estimate.